src/modules/consumer.js

```diff
@@ -108,40 +108,7 @@ class Consumer {
             params: { queue: q.queue },
           });
 
-          channel.consume(
-            q.queue,
-            (msg) => {
-              const messageString = msg.content.toString();
-              this._connection.config.transport.debug(loggerAlias, `[${q.queue}] < ${messageString}`);
-              this._connection.config.logger.debug({
-                message: `${loggerAlias} [${q.queue}] < ${messageString}`,
-                params: { queue: q.queue, message: messageString },
-              });
-
-              // main answer management chaining
-              // receive message, parse it, execute callback, check if should answer, ack/reject message
-              Promise.resolve(parsers.in(msg))
-                .then((body) => callback(body, msg.properties))
-                .then(this.checkRpc(msg, q.queue))
-                .then(() => {
-                  channel.ack(msg);
-                })
-                .catch((error) => {
-                  // if something bad happened in the callback, reject the message so we can requeue it (or not)
-                  this._connection.config.transport.error(loggerAlias, error);
-                  this._connection.config.logger.error({
-                    message: `${loggerAlias} Failed processing message from queue ${q.queue}: ${error.message}`,
-                    error,
-                    params: { queue: q.queue, message: messageString },
-                  });
-
-                  channel.reject(msg, this._connection.config.requeue);
-                });
-            },
-            { noAck: false }
-          );
-
-          return true;
+          return this._consumeQueue(channel, q.queue, callback);
```

Comment:
Before the promise wasn't returned which means we didn't retry the "channel.consume" part if it failed, do we want to change this behaviour intentionally?

Reply:
The `channel.consume` part has a `catch` in it so it never throws, and doesn't retry on it's own.
When the `getChannel` succeeds but the following actions on the `consume` fails, what triggers the retry isn't the `catch` callback, but the "channel closed" event listener.

The retry on the `catch` is for when the `getChannel` failed and we didn't get to the point of adding the listener.

PS as I'm writing this comment I see the `assertQueue` part could still throw after the `channel.addListener('close', ...)`, which could cause multiple retries and duplicate channels, so I'll add a catch there as well and clean the code more.